### PR TITLE
Adding new Automation Schema files

### PR DESCRIPTION
**Pull request by vrdmr**
Adding new schema files for Automation Resources.

### CODE REVIEW

**Comment by vrdmr** [comment]
@eshaparmar FYI

**Comment by vrdmr** [comment]
Hi @TianoMS - When you get a chance, could you please take a look at this PR. Thanks.

**Comment by eshaparmar** [comment]
@TianoMS can you please review this? We are tracking this fix for a CRI and a customer is waiting for this.

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 1]
```diff
@@ -241,15 +241,17 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/automationAccounts_connections" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/automationAccounts_variables" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/automationAccounts_schedules" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/automationAccounts_jobs" },
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: >automationAccounts_jobs [](start = 131, length = 23)

why are these removed?

File: schemas/2015-10-31/Microsoft.Automation.json [hunk 2]
```diff
@@ -147,70 +138,6 @@
       ],
       "description": "Microsoft.Automation/automationAccounts/certificates"
     },
-    "automationAccounts_compilationjobs": {
```
Comment (TianoMS) [inline schemas/2015-10-31/Microsoft.Automation.json]: >automationAccounts_compilationjobs [](start = 5, length = 34)

removing these is a breaking change. why do you remove them?

File: schemas/2014-04-01-preview/deploymentTemplate.json (hunk 1 above)
Comment (Tiano2017) [reply schemas/2014-04-01-preview/deploymentTemplate.json]: as long as the old api-version is still supported, I think you should keep them in schema.

Comment (vrdmr) [reply schemas/2014-04-01-preview/deploymentTemplate.json]: Understood. Fixed, and send a new commit.

File: schemas/2015-10-31/Microsoft.Automation.json (hunk 2 above)
Comment (vrdmr) [reply schemas/2015-10-31/Microsoft.Automation.json]: Understood. Fixed, and sent a new commit.

**Comment by Tiano2017** [comment]
@eshaparmar  @vrdmr  a few resource types are being removed in 2015-10-31/Microsoft.Automation.json. we can't merge the change until we get a justification for it.
